### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by AugustAasvaer** [Raoolo/computational_intelligence#1]
## Review of lab 2

- Firstly it would have been easier to evaluate the code if the readme.txt file had included some information about the techniques applied in the ES. For example, the select_strategy() function seems to be implemented using a roulette wheel technique for choosing the strategy.

- The code quality itself is good across the board. However, there are some optimizations that could be made to make it run faster. For example, the normalize() method which seeks to normalize a vector to the value of 1, could use a more efficient library method like sklearn.processing.normalize() or be implemented more efficiently using the list operations of numpy. 

- The evolving strategy in the solutions is a meta-strategy. This means that the evolving strategy is using the already implemented strategies (pure-random, optimal, etc.), based on a floating point array that gives the probabilities for the individual of choosing one of these pre-defined strategies. Assuming that the ES and the optimal strategy are implemented correctly the best solution should always converge towards the predefined optimal, given enough iterations, as the probability of choosing the optimal tends towards 1. In a realistic case, it would not make sense to apply an ES to a problem where there already exists a known optimal one. Based on this representation the space of strategies is highly limited as it can only represent a permutation of moves, using the predefined ones. In a case where the optimal solution would not exist in the predefined strategies, this might limit the ability to find a sufficient strategy. Therefore I would recommend changing the representation of the problem in such a way that it is not dependent on the predefined solutions. This can increase the space of different strategies and therefore help find a better one in a realistic problem.
